refactor: Implement HeaderLink component

TomasEng opened the pull request:
## Description
When replacing `asChild` with `as`, we need alternative ways to render anchor components from React Router. This pull request creates a `HeaderLink` component for that purpose and replaces the main navigation links in Studio with this component.

## Related Issue
- #14006

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)

standeren commented on frontend/libs/studio-components/src/components/StudioPageHeader/StudioPageHeaderButton/StudioPageHeaderButton.tsx:
```diff
@@ -12,17 +12,10 @@ export type StudioPageHeaderButtonProps = {
       <StudioButton
         ref={ref}
-        className={cn(classes.button, givenClass, getClassName())}
+        className={cn(classes.button, classes[variant], classes[color], givenClass)}
```
nice 😎

standeren commented on frontend/libs/studio-components/src/components/StudioPageHeader/StudioPageHeaderLink/StudioPageHeaderLink.tsx:
```diff
@@ -0,0 +1,31 @@
+  renderLink: (props: HTMLAttributes<HTMLAnchorElement>) => ReactElement;
+} & HTMLAttributes<HTMLAnchorElement>;
+
+export function StudioPageHeaderLink({
```
You have probably already thought about it, but is it not more precisely calling this component `StudioPageHeaderHeaderLink`? But I see that its a mouthfull 😅

TomasEng replied:
I agree. I didn't change it because it would require changing `StudioPageHeader` as well, but I've made changes there anyways. I'll fix.

standeren commented on frontend/libs/studio-components/src/components/StudioPageHeader/common.module.css:
```diff
@@ -0,0 +1,40 @@
+.linkOrButton {
+  --studio-button-regular-dark: #393b51;
+  --studio-button-regular-medium: #3e4f62;
+  --studio-button-regular-light: #657281;
+
+  --studio-button-preview-dark: #0c6536;
+  --studio-button-preview-medium: #118849;
+  --studio-button-preview-light: #67b38a;
```
Is there no FDS variables for these?

TomasEng replied:
Apparently not. Maybe we should centralize Studio-specific colour variables somewhere, but that'd be another issue. For now it's best to define them close to where they're used.

standeren reviewed (approved):
Looks good! 🚀

TomasEng commented:
I have now changed the names as you suggested, @standeren. I discovered that I forgot to remove the CSS file for the `HeaderButton` component and update the reference, so I've fixed that as well.

standeren reviewed (approved):
Perfect 👏

ErlingHauan reviewed (commented):
## Testing
Works as expected 😊 

One thing I noticed is that the menu links have increased in size. I don't mind, just wanted to let you know, in case it wasn't intentional.

Your PR is on the left:
![bilde](https://github.com/user-attachments/assets/1f7cf2ca-96cc-4274-875d-58a74097445b)

ErlingHauan commented:
> Good catch, @ErlingHauan! The links are no longer built on the `Button` component, so they inherit their size from the parent element. I'd argue that this is cleaner, since these links are not supposed to imitate the style of a button anyways. Also it makes sense that they are larger, since this is the main navigation bar.

I can agree with that!

Maybe @Annikenkbrathen wants to review this as well?

Annikenkbrathen commented:
Yes, as Erling said, there is a slight size difference between the link-buttons in the top header and the button in the line below, but personally, I don’t think it matters. They just look a bit more important than the ones below, which is true. Looks good to me!
